app/javascript/components/server-components/HelpCenter/ArticlesIndexPage.tsx

```
@@ -0,0 +1,93 @@
+
+const escapeRegExp = (s: string) => s.replace(/[.*+?^${}()|[\]\\]/gu, "\\$&");
+
+const renderHighlightedText = (text: string, searchTerm: string): React.ReactNode => {
+  if (!searchTerm) return text;
+
+  const escaped = escapeRegExp(searchTerm);
+  const regex = new RegExp(`(${escaped})`, "giu");
+
+  return (
+    <span
+      dangerouslySetInnerHTML={{
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)